internal/generated/api/retry_http_client.go

```diff
@@ -0,0 +1,332 @@
+//
+// Request Body Handling:
+// The transport buffers request bodies to enable retries of non-idempotent requests.
+// This ensures that POST, PUT, and other requests with bodies can be safely retried.
+type RetryTransport struct {
+	// Base is the underlying RoundTripper to use for making requests.
+	// If nil, http.DefaultTransport is used.
+	Base http.RoundTripper
+}
+
+// base returns the underlying RoundTripper to use for making HTTP requests.
+// If Base is nil, it returns http.DefaultTransport as a fallback.
+func (t *RetryTransport) base() http.RoundTripper {
+	if t.Base != nil {
+		return t.Base
+	}
+	return http.DefaultTransport
+}
+
+// baseBackoffDelay is the initial delay used as the base for exponential backoff calculations.
+// The first retry will wait approximately this duration (plus jitter).
+// Value: 1 second
+const baseBackoffDelay = time.Second * 1
+
+// maxBackoffDelay is the maximum delay that will be applied between retry attempts.
+// This prevents exponentially growing delays from becoming excessively long.
+// Even with jitter, delays will not exceed this value by more than 25%.
+// Value: 30 seconds
+const maxBackoffDelay = time.Second * 30
+
+// maxRetryAttempts defines the maximum number of retry attempts that will be made
+// for retryable HTTP responses (429 and 504). After this many failed attempts,
+// the last response will be returned to the caller without further retries.
+// Value: 5 attempts (6 total requests including the initial request)
+const maxRetryAttempts = 5
+
+// calculateBackoff computes the delay duration for retry attempts using exponential
+// backoff with jitter. This helps to distribute retry attempts over time and prevents
+// thundering herd problems when multiple clients are retrying simultaneously.
+//
+// Parameters:
+//   - attempt: The retry attempt number (0-based). The first retry is attempt 0,
+//     the second retry is attempt 1, etc.
+//
+// Returns:
+//   - A time.Duration representing how long to wait before the next retry attempt.
+//
+// Algorithm:
+//  1. Calculate base delay: baseBackoffDelay * 2^attempt
+//  2. Cap the delay at maxBackoffDelay to prevent excessively long waits
+//  3. Add random jitter of ±25% to prevent synchronized retries
+//  4. Ensure the result is never negative
+//
+// Examples:
+//   - Attempt 0: ~1s (750ms - 1.25s with jitter)
+//   - Attempt 1: ~2s (1.5s - 2.5s with jitter)
+//   - Attempt 2: ~4s (3s - 5s with jitter)
+//   - Attempt 3: ~8s (6s - 10s with jitter)
+//   - Attempt 4: ~16s (12s - 20s with jitter)
+//   - Attempt 5+: ~30s (22.5s - 37.5s with jitter, capped at maxBackoffDelay)
+func calculateBackoff(attempt int) time.Duration {
+	// Calculate exponential backoff: baseDelay * 2^attempt
+	delay := float64(baseBackoffDelay) * math.Pow(2, float64(attempt))
+
+	// Cap the delay at maxBackoffDelay
+	if delay > float64(maxBackoffDelay) {
+		delay = float64(maxBackoffDelay)
+	}
```

Comment:
nit

can do 

`delay = min(delay, maxBackoffDelay)`

Reply:
what is min? another user defined function? why would we want to define it, if it's used only at one place?